third-party/rsutils/include/rsutils/easylogging/easyloggingpp.h

```diff
@@ -41,17 +40,63 @@
+    do                                                                                                                 \
+    {                                                                                                                  \
+        auto logger__ = el::Loggers::getLogger( rsutils::g_librealsense_elpp_id );                                     \
+        if( logger__ && logger__->enabled( el::Level::LEVEL ) )                                                        \
+        {                                                                                                              \
+            el::base::Writer( el::Level::LEVEL, __FILE__, __LINE__, ELPP_FUNC, el::base::DispatchAction::NormalLog )   \
+                    .construct( logger__ )                                                                             \
+                << STR;                                                                                                \
+        }                                                                                                              \
+    }                                                                                                                  \
+    while( false )
+
+// Same, but first convert to string using usual mechanisms
+#define LIBRS_LOG_( LEVEL, ... )                                                                                       \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        auto logger__ = el::Loggers::getLogger( rsutils::g_librealsense_elpp_id );                                     \
+        if( logger__ && logger__->enabled( el::Level::LEVEL ) )                                                        \
+        {                                                                                                              \
+            std::ostringstream os__;                                                                                   \
+            os__ << __VA_ARGS__;                                                                                       \
+            el::base::Writer( el::Level::LEVEL, __FILE__, __LINE__, ELPP_FUNC, el::base::DispatchAction::NormalLog )   \
+                    .construct( logger__ )                                                                             \
+                << os__.str();                                                                                         \
+        }                                                                                                              \
+    }                                                                                                                  \
+    while( false )
+
+#define LOG_DEBUG(...)    LIBRS_LOG_( Debug,    __VA_ARGS__ )
```

Comment:
Can you add a comment about each group?
When should the developer use what?

Reply:
It says, line 51:
```
// Direct log to ELPP, without conversion to string first
```
And line 66:
```
// Same, but first convert to string using usual mechanisms
```
Usually you don't need to think about it. Just use the regular macros and everything will work fine. I just added extra flexibility when you do want to optimize.